### PR TITLE
[6CORE-043] Create ER diagram and SQL Table for Warehouse Service

### PR DESCRIPTION
**Pull request by YuraKhilko**
Added DDL scripts and ER-diagram for warehouse microservice. Added docker command for launching Postgres DB to README.md

### CODE REVIEW

**Review by kateryna-mykh — COMMENTED** [review]
Would be great to add the picture representation of the diagram.

**Review by kateryna-mykh — APPROVED** [review]

File: 3-platform/warehouse/pom.xml [hunk 1]
```diff
@@ -19,6 +19,7 @@
     <module>app</module>
     <module>domain</module>
     <module>spec</module>
+    <module>db_postgres</module>
```
Comment (sokimaaa) [inline 3-platform/warehouse/pom.xml]: it is not maven module. so no need to have it here
```suggestion
```

File: 3-platform/warehouse/db_postgres/src/main/resources/db/changelog/db.changelog-master.yaml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+databaseChangeLog:
```
Comment (sokimaaa) [inline 3-platform/warehouse/db_postgres/src/main/resources/db/changelog/db.changelog-master.yaml]: Is it asked to have liquibase in the scope of this task?

File: 3-platform/warehouse/db_postgres/src/main/resources/db/changelog/changes/02-create-products-table.sql [hunk 3]
```diff
@@ -0,0 +1,21 @@
+-- products table
+
+CREATE TABLE public.products (
+                                 id varchar NOT NULL,
+                                 "name" varchar NOT NULL,
+                                 description varchar NULL,
+                                 image varchar NULL,
+                                 price bigint NOT NULL,
+                                 category varchar NULL,
+                                 CONSTRAINT products_pk PRIMARY KEY (id),
```
Comment (sokimaaa) [inline 3-platform/warehouse/db_postgres/src/main/resources/db/changelog/changes/02-create-products-table.sql]: `CTRL + ALT + O` and `CTRL + ALT + L` are hot keys for format fix.

File: 3-platform/warehouse/db_postgres/pom.xml [hunk 4]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
```
Comment (sokimaaa) [inline 3-platform/warehouse/db_postgres/pom.xml]: why we need this?

File: 3-platform/warehouse/db_postgres/pom.xml [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+  <parent>
+    <groupId>com.6core.platform</groupId>
+    <artifactId>warehouse</artifactId>
+    <version>0.0.1-SNAPSHOT</version>
+  </parent>
+
+  <groupId>org.example</groupId>
+  <artifactId>db_postgres</artifactId>
```
Comment (sokimaaa) [inline 3-platform/warehouse/db_postgres/pom.xml]: For future: dont use underscore *_*
```suggestion
  <artifactId>db_postgres</artifactId>
```

File: 3-platform/warehouse/db_postgres/src/main/resources/db/changelog/changes/01-create-warehouses-table.sql [hunk 6]
```diff
@@ -0,0 +1,15 @@
+-- warehouses table
+
+CREATE TABLE public.warehouses
+(
+    id      varchar NOT NULL,
```
Comment (sokimaaa) [inline 3-platform/warehouse/db_postgres/src/main/resources/db/changelog/changes/01-create-warehouses-table.sql]: NOT NULL for Primary key? could it have been null?

Comment (sokimaaa) [reply 3-platform/warehouse/db_postgres/src/main/resources/db/changelog/changes/01-create-warehouses-table.sql]: the same for FK

Comment (sokimaaa) [reply 3-platform/warehouse/db_postgres/src/main/resources/db/changelog/changes/01-create-warehouses-table.sql]: BTW: foreign key could be null for some cases. 

e.g. inventory -> product. 

Inventory is allocated, but product is not assigned.

File: 3-platform/warehouse/db_postgres/src/main/resources/db/changelog/changes/02-create-products-table.sql [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+                                 CONSTRAINT products_price_check CHECK (price>=0)
+);
+
+-- Column comments
+
+COMMENT ON COLUMN public.products.id IS 'PK of product record';
```
Comment (sokimaaa) [inline 3-platform/warehouse/db_postgres/src/main/resources/db/changelog/changes/02-create-products-table.sql]: pls follow up that other team members align with your style of sql.

Comment (sokimaaa) [reply 3-platform/warehouse/db_postgres/src/main/resources/db/changelog/changes/02-create-products-table.sql]: e.g. using Comment ON COLLUMN

File: 3-platform/warehouse/db_postgres/src/main/resources/db/changelog/changes/03-create-inventories-table.sql [hunk 8]
```diff
@@ -0,0 +1,24 @@
+-- inventories table
+
+CREATE TABLE public.inventories
```
Comment (sokimaaa) [inline 3-platform/warehouse/db_postgres/src/main/resources/db/changelog/changes/03-create-inventories-table.sql]: add precondition e.g.` if not exist`

File: 3-platform/warehouse/db_postgres/src/main/resources/db/changelog/changes/03-create-inventories-table.sql [hunk 9]
```diff
@@ -0,0 +1,24 @@
+-- inventories table
+
```
Comment (sokimaaa) [inline 3-platform/warehouse/db_postgres/src/main/resources/db/changelog/changes/03-create-inventories-table.sql]: move sql scripts to the warehouses/db/postgres/

we are not going to store it in resources

File: 3-platform/warehouse/db_postgres/src/main/resources/db/changelog/changes/03-create-inventories-table.sql [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+    warehouse_id       varchar NOT NULL,
+    product_id         varchar NULL,
+    actual_quantity    int     NULL,
+    available_quantity int     NULL,
+    CONSTRAINT inventories_pk PRIMARY KEY (id),
+    CONSTRAINT inventories_actual_quantity_check CHECK (actual_quantity >= 0),
+    CONSTRAINT inventories_available_quantity_check CHECK (available_quantity >= 0),
+    CONSTRAINT inventories_check CHECK (available_quantity <= actual_quantity),
+    CONSTRAINT inventories_warehouses_fk FOREIGN KEY (id) REFERENCES public.warehouses (id),
+    CONSTRAINT inventories_products_fk FOREIGN KEY (id) REFERENCES public.products (id)
```
Comment (sokimaaa) [inline 3-platform/warehouse/db_postgres/src/main/resources/db/changelog/changes/03-create-inventories-table.sql]: what do you think about adding these constraints (except PK, FK) separately via `alter table add constraints`?
```suggestion
    CONSTRAINT inventories_actual_quantity_check CHECK (actual_quantity >= 0),
    CONSTRAINT inventories_available_quantity_check CHECK (available_quantity >= 0),
    CONSTRAINT inventories_check CHECK (available_quantity <= actual_quantity),
    CONSTRAINT inventories_warehouses_fk FOREIGN KEY (id) REFERENCES public.warehouses (id),
    CONSTRAINT inventories_products_fk FOREIGN KEY (id) REFERENCES public.products (id)
```

**Review by sokimaaa — CHANGES_REQUESTED** [review]

File: 3-platform/warehouse/pom.xml (hunk 1 above)
Comment (YuraKhilko) [reply 3-platform/warehouse/pom.xml]: I've just thought that point from the task "The submodule db/postgres is created" means to create new module. Then what does that point mean?

File: 3-platform/warehouse/db_postgres/src/main/resources/db/changelog/changes/02-create-products-table.sql (hunk 3 above)
Comment (YuraKhilko) [reply 3-platform/warehouse/db_postgres/src/main/resources/db/changelog/changes/02-create-products-table.sql]: thanks, missed that.

File: 3-platform/warehouse/db_postgres/src/main/resources/db/changelog/changes/02-create-products-table.sql (hunk 7 above)
Comment (YuraKhilko) [reply 3-platform/warehouse/db_postgres/src/main/resources/db/changelog/changes/02-create-products-table.sql]: It's possible to create a table using IDE and after that generate DDL script. And "COMMENT" - it's IDE's creature.
So, let's don't make our colleagues change uppercase.

File: 3-platform/warehouse/db_postgres/src/main/resources/db/changelog/db.changelog-master.yaml (hunk 2 above)
Comment (YuraKhilko) [reply 3-platform/warehouse/db_postgres/src/main/resources/db/changelog/db.changelog-master.yaml]: Nope. Have deleted.

File: 3-platform/warehouse/db_postgres/pom.xml (hunk 4 above)
Comment (YuraKhilko) [reply 3-platform/warehouse/db_postgres/pom.xml]: have deleted.

File: 3-platform/warehouse/pom.xml (hunk 1 above)
Comment (YuraKhilko) [reply 3-platform/warehouse/pom.xml]: Maybe, that point means folder db/postgres is created.

File: 3-platform/warehouse/db_postgres/src/main/resources/db/changelog/changes/01-create-warehouses-table.sql (hunk 6 above)
Comment (YuraKhilko) [reply 3-platform/warehouse/db_postgres/src/main/resources/db/changelog/changes/01-create-warehouses-table.sql]: No, PK can't be null, but for my surprising IDE generate that constraint. Should I delete it?

File: 3-platform/warehouse/db_postgres/src/main/resources/db/changelog/changes/03-create-inventories-table.sql (hunk 10 above)
Comment (YuraKhilko) [reply 3-platform/warehouse/db_postgres/src/main/resources/db/changelog/changes/03-create-inventories-table.sql]: Ok, let's separate that checks and paste them after table creation.

File: 3-platform/warehouse/db_postgres/src/main/resources/db/changelog/changes/01-create-warehouses-table.sql (hunk 6 above)
Comment (YuraKhilko) [reply 3-platform/warehouse/db_postgres/src/main/resources/db/changelog/changes/01-create-warehouses-table.sql]: I think it's better to leave that not null constraint for PK, just as agreement that every field should have that parameter (null/not null)

**Comment by YuraKhilko** [comment]
> Would be great to add the picture representation of the diagram.

@sokimaaa is it possible to add picture of ER diagram just to be able to check it by eyes? Or it's supposed that reviewer copy past it local and look throw generated picture?

File: 3-platform/warehouse/db_postgres/src/main/resources/db/changelog/db.changelog-master.yaml (hunk 2 above)
Comment (sokimaaa) [reply 3-platform/warehouse/db_postgres/src/main/resources/db/changelog/db.changelog-master.yaml]: I still see this file. Did you push changes?

**Comment by sokimaaa** [comment]
` is it possible to add picture of ER diagram just to be able to check it by eyes? Or it's supposed that reviewer copy past it local and look throw generated picture?` -> Check how it was done for sequence diagrams.

**Review by sokimaaa — CHANGES_REQUESTED** [review]

File: 3-platform/warehouse/db_postgres/src/main/resources/db/changelog/db.changelog-master.yaml (hunk 2 above)
Comment (YuraKhilko) [reply 3-platform/warehouse/db_postgres/src/main/resources/db/changelog/db.changelog-master.yaml]: Not yet.

**Review by sokimaaa — APPROVED** [review]

**Pull request merged**